fix: username for Notification Log was wrong for Administrator

stephenBDT opened the pull request:
There is a problem for the user 'Administrator' which has the property, that the document name for that User-document is NOT the email. For all other users, the email is also the docname for the User-document.

The problem is, that notification log tries to create a notification for the "User" with the administrator email, but then frappe complains that this user can not be found.

This is of course the case because that Users name is not his/her email, but 'Administrator'.

This fixes that

stephenBDT committed:
fix: username for Notification Log was wrong for Administrator

ankush commented on frappe/desk/doctype/notification_log/notification_log.py:
```diff
@@ -63,15 +63,16 @@ def make_notification_logs(doc, users):
 
 	for user in users:
-		if frappe.db.exists('User', {"email": user, "enabled": 1}):
+		username = frappe.db.exists("User", {"email": user, "enabled": 1})
```
```suggestion
		username = frappe.db.get_value("User", {"email": user, "enabled": 1})
```
When the value is used for anything other than existence check it should be `get_value` for correct semantics.

ankush committed:
refactor: simplify make_notification_logs

- Split overindented code
- Remove check for EPS, if EPS is disabled then why is it creating
  notification?

ankush commented:
_had_ to rebase, last commit was from 2021. 

Don't merge yet, found similar misuse in other places.

ankush committed:
fix: fetch email ID from User document before notifying
